src/snowcli/cli/nativeapp/init.py

```diff
@@ -238,76 +223,115 @@ def _init_with_url_and_template(
-        if path_to_project:
-            rmtree(path_to_project.resolve())
+        if project_path.exists():
+            rmtree(project_path.resolve())
 
         log.error(err)
         raise InitError()
 
 
+def _is_part_of_existing_project(path: Path):
+    project_file = path / "snowflake.yml"
+    if project_file.is_file():
+        # Found a snowflake.yml in an ancestor directory, this path is part of an existing project
+        return True
+
+    parent = path.parent
+    if parent == path:
+        # reached the root without encountering an existing project
+        return False
+
+    return _is_part_of_existing_project(parent)
+
+
 def nativeapp_init(
-    name: str, git_url: Optional[str] = None, template: Optional[str] = None
+    path: str,
+    name: Optional[str] = None,
+    git_url: Optional[str] = None,
+    template: Optional[str] = None,
 ):
     """
     Initialize a Native Apps project in the user's current working directory, with or without the use of a template.
 
     Args:
+        path (str): The location of the project to be created.
         name (str): Name of the project to be created.
         git_url (str): The git URL to perform a clone from.
         template (str): A template within the git URL to use, all other directories and files outside the template will be discarded.
 
     Returns:
         None
     """
+    try:
+        project_path = Path(path).expanduser().resolve()
+    except Exception as err:  # expanduser can fail
+        raise InitError()
 
-    current_working_directory = Path.cwd()
+    # If a subdirectory with the same name as name exists in the current directory, fail init command
+    if project_path.exists():
+        raise DirectoryAlreadyExistsError(path)
+
+    # Check if the specified path already exists in a native apps project. Nesting projects is not allowed.
+    if _is_part_of_existing_project(project_path.parent):
+        raise CannotInitializeAnExistingProjectError()
 
-    # If the intended project name is not a valid identifier, fail init command
-    if not is_valid_project_name(name):
+    project_name = (
+        name if name is not None else _generate_project_name_from_path(project_path)
+    )
+    if not project_name:
+        # empty name
         raise ProjectNameInvalidError()
 
-    # If current directory is already contains a file named snowflake.yml, i.e. is a native apps project, fail init command.
-    # We do not validate contents of the yml here though.
-    path_to_snowflake_yml = current_working_directory / "snowflake.yml"
-    if path_to_snowflake_yml.is_file():
-        raise CannotInitializeAnExistingProjectError()
+    if not is_valid_identifier(project_name) and (
+        project_name.startswith('"') or project_name.endswith('"')
+    ):
+        # the project name looks like it was partially double quoted. This is likely a mistake, reject it rather than
+        # silently escaping it.
+        raise ProjectNameInvalidError()
```

Comment:
Possible to add the invalid project name to the args of the exception so users can see it in the error message?